File: EE3900-main/filter/codes/3_3.c
#include<stdio.h>
#include<stdlib.h>
int main()
{
    FILE* f = NULL;
    f = fopen("data.dat","w");
    double x[6] ={1.0,2.0,3.0,4.0,2.0,1.0};
    int k=20; 
    double y[20]={0.0};
    y[0] = x[0];
    y[1] = -0.5*y[0]+x[1];


    for(int n=2;n<k-1;n++)
    {
        if (n < 6)
		    y[n] = -0.5*y[n-1]+x[n]+x[n-2];
	    else if (n > 5 && n < 8)
		    y[n] = -0.5*y[n-1]+x[n-2];
	    else
		    y[n] = -0.5*y[n-1];
    }  
    for(int i=0;i<k-1;i++)
    {
        fprintf(f,"%lf ",y[i]);
    } 
}

File: EE3900-main/filter/codes/3_3.py
import numpy as np
import matplotlib.pyplot as plt

X = np.array([1.0,2.0,3.0,4.0,2.0,1.0])

Y = np.loadtxt("codes/data.dat",dtype = np.double)
print(Y)

plt.subplot(2,1,1)
plt.stem(range(0,6),X)
plt.xlabel('$n$')
plt.ylabel('$X(n)$')

plt.subplot(2,1,2)
plt.stem(range(0,19),Y)
plt.xlabel('$n$')
plt.ylabel('$Y(n)$')

plt.grid()
plt.show()